diff --git a/libwww2/HTFWriter.c b/libwww2/HTFWriter.c
index dbd6233..5416b67 100644
--- a/libwww2/HTFWriter.c
+++ b/libwww2/HTFWriter.c
@@ -513,7 +513,7 @@ PUBLIC HTStream* HTSaveAndExecute ARGS5(
                 ((strlen (pres->command) + 10 + 3*strlen(me->fnam)) * 
                  sizeof (char));
               
-              /* Cute.  pres->command will be something like "xv %s"; me->fnam
+              /* Cute.  pres->command will be something like "sxiv %s"; me->fnam
                  gets filled in as many times as appropriate.  */
               sprintf (command, pres->command, me->fnam, me->fnam, me->fnam);
               
diff --git a/libwww2/HTInit.c b/libwww2/HTInit.c
index 928a9cb..fa4c040 100644
--- a/libwww2/HTInit.c
+++ b/libwww2/HTInit.c
@@ -92,19 +92,19 @@ PUBLIC void HTFormatInit NOARGS
 #endif /* not ultrix or __alpha */
 #endif /* not __sgi */
 
-      HTSetPresentation("image/gif", "xv %s", 1.0, 3.0, 0.0);
-      HTSetPresentation("image/jpeg", "xv %s", 1.0, 3.0, 0.0);
-      HTSetPresentation("image/png", "xv %s", 1.0, 3.0, 0.0);
-      HTSetPresentation("image/x-png", "xv %s", 1.0, 3.0, 0.0);
-      HTSetPresentation("image/tiff", "xv %s", 1.0, 3.0, 0.0);
-      HTSetPresentation("image/x-portable-anymap", "xv %s", 1.0, 3.0, 0.0);
-      HTSetPresentation("image/x-portable-bitmap", "xv %s", 1.0, 3.0, 0.0);
-      HTSetPresentation("image/x-portable-graymap", "xv %s", 1.0, 3.0, 0.0);
-      HTSetPresentation("image/x-portable-pixmap", "xv %s", 1.0, 3.0, 0.0);
-      HTSetPresentation("image/x-rgb", "xv %s", 1.0, 3.0, 0.0);
-      HTSetPresentation("image/rgb", "xv %s", 1.0, 3.0, 0.0);
-      HTSetPresentation("image/x-xbitmap", "xv %s", 1.0, 3.0, 0.0);
-      HTSetPresentation("image/x-xpixmap", "xv %s", 1.0, 3.0, 0.0); /* ?? */
+      HTSetPresentation("image/gif", "sxiv %s", 1.0, 3.0, 0.0);
+      HTSetPresentation("image/jpeg", "sxiv %s", 1.0, 3.0, 0.0);
+      HTSetPresentation("image/png", "sxiv %s", 1.0, 3.0, 0.0);
+      HTSetPresentation("image/x-png", "sxiv %s", 1.0, 3.0, 0.0);
+      HTSetPresentation("image/tiff", "sxiv %s", 1.0, 3.0, 0.0);
+      HTSetPresentation("image/x-portable-anymap", "sxiv %s", 1.0, 3.0, 0.0);
+      HTSetPresentation("image/x-portable-bitmap", "sxiv %s", 1.0, 3.0, 0.0);
+      HTSetPresentation("image/x-portable-graymap", "sxiv %s", 1.0, 3.0, 0.0);
+      HTSetPresentation("image/x-portable-pixmap", "sxiv %s", 1.0, 3.0, 0.0);
+      HTSetPresentation("image/x-rgb", "sxiv %s", 1.0, 3.0, 0.0);
+      HTSetPresentation("image/rgb", "sxiv %s", 1.0, 3.0, 0.0);
+      HTSetPresentation("image/x-xbitmap", "sxiv %s", 1.0, 3.0, 0.0);
+      HTSetPresentation("image/x-xpixmap", "sxiv %s", 1.0, 3.0, 0.0); /* ?? */
 
       HTSetPresentation("image/xwd", "xwud -in %s", 1.0, 3.0, 0.0);
       HTSetPresentation("image/x-xwd", "xwud -in %s", 1.0, 3.0, 0.0);
