diff --git a/src/Config.cpp b/src/Config.cpp
index ecc1fce..b2fd9bb 100644
--- a/src/Config.cpp
+++ b/src/Config.cpp
@@ -19,14 +19,7 @@ namespace QuickMedia {
     // while others are in this pseudo shell config file ~/.config/user-dirs.dirs
     static std::map<std::string, std::string> get_xdg_variables() {
         std::string user_dirs_filepath;
-        const char *xdg_config_home = getenv("XDG_CONFIG_HOME");
-        if(xdg_config_home) {
-            user_dirs_filepath = xdg_config_home;
-        } else {
-            user_dirs_filepath = get_home_dir().data + "/.config";
-        }
-
-        user_dirs_filepath += "/user-dirs.dirs";
+        user_dirs_filepath = get_home_dir().data + "/.config/user-dirs.dirs";
 
         std::map<std::string, std::string> result;
         FILE *f = fopen(user_dirs_filepath.c_str(), "rb");
diff --git a/src/Storage.cpp b/src/Storage.cpp
index 5c12911..7ab9249 100644
--- a/src/Storage.cpp
+++ b/src/Storage.cpp
@@ -89,11 +89,7 @@ namespace QuickMedia {
 
     Path get_storage_dir() {
         if(xdg_config_home.data.empty()) {
-            const char *xdg_config_home_p = getenv("XDG_CONFIG_HOME");
-            if(xdg_config_home_p)
-                xdg_config_home = xdg_config_home_p;
-            else
-                xdg_config_home = get_home_dir().join(".config");
+            xdg_config_home = get_home_dir().join(".config");
         }
 
         Path storage_dir = xdg_config_home;
diff --git a/src/VideoPlayer.cpp b/src/VideoPlayer.cpp
index 3aa4eca..277271a 100644
--- a/src/VideoPlayer.cpp
+++ b/src/VideoPlayer.cpp
@@ -137,12 +137,7 @@ namespace QuickMedia {
     }
 
     static Path get_config_dir_xdg() {
-        Path path;
-        const char *xdg_config_home_p = getenv("XDG_CONFIG_HOME");
-        if(xdg_config_home_p)
-            path = xdg_config_home_p;
-        else
-            path = get_home_dir().join(".config");
+        Path path = get_home_dir().join(".config");
         return path;
     }
 
