--- a/src/plugins/Matrix.cpp	2024-01-02 01:57:58.399694763 +0000
+++ b/src/plugins/Matrix.cpp	2024-01-02 01:59:28.217714634 +0000
@@ -227,13 +227,13 @@
         string_replace_all(result, '\r', ' ');
         size_t index = result.find('\n');
         if(index == std::string::npos) {
-            if(result.size() > max_length)
-                return result.substr(0, max_length) + "...";
+            if(result.size() > (max_length*4))
+                return result.substr(0, (max_length*4)) + "...";
             return result;
         } else if(index == 0) {
             return "";
         } else {
-            return result.substr(0, std::min(index, max_length)) + "...";
+            return result.substr(0, std::min(index, (max_length*4))) + "...";
         }
     }
 
@@ -6199,4 +6199,4 @@
                 break;
         }
     }
-}
\ No newline at end of file
+}
