diff --git a/paludis/repositories/e/ebuild/utils/strip_tar_corruption.cc b/paludis/repositories/e/ebuild/utils/strip_tar_corruption.cc
index 221051072..c9a3fbc85 100644
--- a/paludis/repositories/e/ebuild/utils/strip_tar_corruption.cc
+++ b/paludis/repositories/e/ebuild/utils/strip_tar_corruption.cc
@@ -21,6 +21,7 @@
 #include <sstream>
 #include <string>
 #include <cstdlib>
+#include <cstdint>
 
 int main(int, char * argv[])
 {
