--- a/xautocfg.cpp
+++ b/xautocfg.cpp
@@ -20,6 +20,7 @@
 #include <unordered_map>
 #include <unistd.h>
 #include <sys/wait.h>
+#include <filesystem>
 
 #include <X11/XKBlib.h>
 #include <X11/Xlib.h>
@@ -97,7 +98,13 @@
 			std::cout << "HOME env not set, can't locate config." << std::endl;
 			exit(1);
 		}
-		cfgpathss << home << "/.config/xautocfg.cfg";
+                std::string homeConfigPath = std::string(home) + "/.config/xautocfg.cfg";
+                
+                if (std::filesystem::exists(homeConfigPath)) {
+                    cfgpathss << homeConfigPath;
+                } else {
+                    cfgpathss << "/etc/xautocfg.cfg";
+                }
 		ret.config = std::move(cfgpathss).str();
 	}
 
